src/Geta.NotFoundHandler/Infrastructure/Initialization/Upgrader.cs

```diff
@@ -5,67 +5,72 @@
 using Geta.NotFoundHandler.Infrastructure.Configuration;
 using Microsoft.Extensions.Logging;
 
-namespace Geta.NotFoundHandler.Infrastructure.Initialization
+namespace Geta.NotFoundHandler.Infrastructure.Initialization;
+
+public class Upgrader
 {
-    public class Upgrader
+    private const string SuggestionsTable = "[dbo].[NotFoundHandler.Suggestions]";
+    private const string RedirectsTable = "[dbo].[NotFoundHandler.Redirects]";
+    private const string RegexRedirectsTable = "[dbo].[NotFoundHandler.RegexRedirects]";
+    private readonly IDataExecutor _dataExecutor;
+    private readonly ILogger<Upgrader> _logger;
+
+    public Upgrader(
+        ILogger<Upgrader> logger,
+        IDataExecutor dataExecutor)
     {
-        private readonly ILogger<Upgrader> _logger;
-        private readonly IDataExecutor _dataExecutor;
-
-        private const string SuggestionsTable = "[dbo].[NotFoundHandler.Suggestions]";
-        private const string RedirectsTable = "[dbo].[NotFoundHandler.Redirects]";
+        _logger = logger;
+        _dataExecutor = dataExecutor;
+    }
 
-        public Upgrader(
-            ILogger<Upgrader> logger,
-            IDataExecutor dataExecutor)
+    public void Start()
+    {
+        _logger.LogDebug("Initializing NotFoundHandler version check");
+        var version = GetVersionNumber();
+        if (version == NotFoundHandlerOptions.CurrentDbVersion)
         {
-            _logger = logger;
-            _dataExecutor = dataExecutor;
+            return;
         }
 
-        public void Start()
+        _logger.LogDebug("Older version found. Version nr. : {Version}", version);
+
+        if (version == -1)
         {
-            _logger.LogDebug("Initializing NotFoundHandler version check");
-            var version = GetVersionNumber();
-            if (version == NotFoundHandlerOptions.CurrentDbVersion)
-            {
-                return;
-            }
-
-            _logger.LogDebug("Older version found. Version nr. : {Version}", version);
-
-            if (version == -1)
-            {
-                Create();
-            }
-            else
-            {
-                Upgrade();
-            }
+            Create();
         }
-
-        /// <summary>
-        /// Create redirects and suggestions tables and SP for version number
-        /// </summary>
-        private void Create()
+        else
         {
-            var created = CreateRedirectsTable();
+            Upgrade();
+        }
+    }
 
-            if (created)
-            {
-                created = CreateSuggestionsTable();
-            }
+    /// <summary>
+    /// Create redirects and suggestions tables and SP for version number
+    /// </summary>
+    private void Create()
+    {
+        var created = CreateRedirectsTable();
```

Comment:
not sure why we need to check if the previous table was created

Reply:
So that we do not continue if something fails.
P.S. This logic is from the beginning of the project.